src/hooks/useWatchPendingTxs.ts

```diff
@@ -28,20 +29,26 @@ export const useWatchPendingTransactions = ({ address }: { address: string }) =>
   const { nativeCurrency, accountAddress } = useAccountSettings();
 
   const refreshAssets = useCallback(
-    (tx: RainbowTransaction) => {
-      if (tx.type === 'swap') {
-        // update swap assets
-        //swapRefreshAssets(tx.nonce);
-      } else {
-        // fetch assets again
-        fetchUserAssets({
-          address: accountAddress,
+    (_: RainbowTransaction) => {
+      // NOTE: We have two user assets stores right now, so let's invalidate both queries and trigger a refetch
+      const connectedToHardhat = getIsHardhatConnected();
+      queryClient.invalidateQueries(
+        userAssetsQueryKey({
+          address,
           currency: nativeCurrency,
-          connectedToHardhat: false,
-        });
-      }
+          connectedToHardhat,
+        })
+      );
+      queryClient.invalidateQueries(
+        swapsUserAssetsQueryKey({
+          address: address as Address,
+          currency: nativeCurrency,
+          testnetMode: !!connectedToHardhat,
+        })
+      );
+      queryClient.invalidateQueries(nftsQueryKey({ address }));
     },
-    [accountAddress, nativeCurrency]
+    [address, nativeCurrency]
```

Comment:
I know this is not related to your code, but I can't figure out when we would ever want to use `accountAddress` over `address` in this hook for the rest of the places it's used in this file - @brunobar79 if you have any thoughts

Reply:
seems like we can just get rid of the address param and use accountAddress instead

Reply:
I think the `address` would be more precise no? like say a user does a transaction and then switches to another wallet - the "watching of the pending txns" should be on the `address` it was originally for, regardless of what the current `accountAddress` happens to be